src/hotspot/share/runtime/handshake.cpp

```diff
@@ -481,12 +480,14 @@ bool HandshakeState::process_self_inner() {
       if (!async) {
         HandleMark hm(_handshakee);
         PreserveExceptionMark pem(_handshakee);
-        op->do_handshake(_handshakee);
+        op->do_handshake(_handshakee); // acquire, op removed after
+        remove_op(op);
       } else {
         // An asynchronous handshake may put the JavaThread in blocked state (safepoint safe).
         // The destructor ~PreserveExceptionMark touches the exception oop so it must not be executed,
         // since a safepoint may be in-progress when returning from the async handshake.
-        op->do_handshake(_handshakee);
+        op->do_handshake(_handshakee); // acquire, op removed after
```

Comment:
Why acquire? Perhaps:
// peek and handshake with op; op removed after

Reply:
It's mean do_handshake provide acquire thus op is removed after.

Reply:
Ok.